src/engine/runner/bootstrapper/express/bootstrapper.ts

```diff
@@ -0,0 +1,35 @@
+import { Cleaner } from "./cleaner";
+import { extractorsFactory } from "./extractors";
+import { ModuleExtractor } from "../../shared/module-extractor";
+import { Configuration, ExpressModule } from "../../../configuration";
+import { Bootstrapper } from "../types";
+
+export class ExpressBootstrapper implements Bootstrapper {
+  private cleaner: Cleaner;
+
+  private extractors: ModuleExtractor[];
+
+  constructor(configuration: Configuration) {
+    const allModules = Object.values(ExpressModule);
```

Comment:
Declaring a variable just to use it once is a bad practice and a way to make codebase slower to read pretty fast.

Reply:
Combining a lot of functions together sometimes is really hard to understand. IMO better to divide it into meaningful functions/variables, to help readers understand a code. Also thanks to that, the code is self documented, so I see pros here.

Reply:
Chaining is built in solution that is commonly used in a lot of languages.
`IMO better to divide it into meaningful functions/variables` yes you are right however, it is also common approach to create anonymouse functions and doing chains.
`Combining a lot of functions together sometimes is really hard to understand.` I agree but I dont see that is a case here. This is what for `compose` function and chaining exists.
Imho this is a really ugly practice to follow. Let me explain.
Lets compare those 2 approaches.
```ts
const test = (data) => Object.fromEntries(Object.entries(data).map(([key, value]) => [key,{key, value}]).filter(([key]) => key.includes("test")));
```
```ts
const test = (data) => {
   const entries = Object.entries(data);
   const entriesMapped = entries.map(([key, value]) => [key,{key, value}]);
   const entriesMappedAndFiltered = entriesMapped.filter(([key]) => key.includes("test"));
   const result = entriesMappedAndFiltered ;
   return result;
}
```
As you can see implementation takes a lot more lines and your codebase is 2 times bigger than if you chaining.
But what if we would add another map to this.
```ts
const test = (data) => Object.fromEntries(Object.entries(data).map(([key, value]) => [key,{key, value}]).filter(([key]) => key.includes("test")).map(([key, value]) => [key,{key, value}]));
```
```ts
const test = (data) => {
   const entries = Object.entries(data);
   const entriesMapped = entries.map(([key, value]) => [key,{key, value}]);
   const entriesMappedAndFiltered = entriesMapped.filter(([key]) => key.includes("test"));
   const entriesMappedAndFilteredAndMappedAgain = entries.map(([key, value]) => [key,{key, value}]);
   const result = entriesMappedAndFilteredAndMappedAgain ;
   return result;
}
```
But what if you will change first map into something else? In approach with redeclaration if you want to stay consistent 
 you will need to maintain also a variable name.
 Also in approach with redeclaration you must discover the right order of functions executions - while in chaining the order is already ensured.
 This is also hard to use when you actually have a variable that is not used only for keeping all states of each process step.
 
 I dont see a way that the doubtful benefit of having all variables available would overcome those cons.
 Do you?

Reply:
Adding variables here helps you maintain the readability of the code because you understand each step and what's going on. If I can help my colleague read my code, and understand it better by introducing variables then I am 100% in.  Maintaining variable names nowadays isn't hard, because we have a lot of beautiful IDE features which help us.
I would also be cautious in judging code depending on the number of lines. Because it tells us nothing about the quality. Sometimes is better to explicitly divide it into smaller, more readable parts than packing it into one big `I am doing everything` function. Because when we leave, somebody else will need to maintain it, and we should help these people.
Anyway, I think that this discussion is too subjective and will not change anything in this one line of code, so let's leave it. If in the future someone finds it unreadable, they can simply change it.